### PR TITLE
Catch error where it set callback incorrectly

### PR DESCRIPTION
**Pull request by pphminions**
#### What does this PR accomplish?
This will catch the error where it set callback incorrectly in migration scripts.

#### How should this be manually tested?
N/A

#### Any background context you want to provide?
N/A

#### What are the relevant Trello cards?
N/A

#### Screenshots (if appropriate)
N/A

### CODE REVIEW

File: test/data/migrations/20141028-aaa.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 exports.up = function(cb) {
   setTimeout(function() {
     cb(null, 'aaa: up');
+    cb(null, 'aaa: up2');
```
Comment (runk) [inline test/data/migrations/20141028-aaa.js]: revert it?

File: lib/migrant.js [hunk 2]
```diff
@@ -72,7 +72,22 @@ Migrant.prototype.migrateSingle = function(mode, filename, cb) {
 
   this.log(filename, mode);
   async.series([
-    migration[mode], // run migration
+    function(cb) {
+      var counter = 0
```
Comment (runk) [inline lib/migrant.js]: just use a flag `called = false` then set to `true`

File: lib/migrant.js [hunk 3]
```diff
@@ -72,7 +72,22 @@ Migrant.prototype.migrateSingle = function(mode, filename, cb) {
...
+      fn = function(err, res) {
+        // to check if the return set correctly on each file.
+        // if there is an error it will be written in the log file and will throw an error.
+        if (counter > 0){
+          var msg = 'The return statmement called more than once in ' + filename;
+          self.log(filename, msg, true);
+          throw new Error('The return statmement called more than once in ' + filename)
```
Comment (runk) [inline lib/migrant.js]: throw new Error(msg)

File: test/data/migrations/20141028-aaa.js (hunk 1 above)
Comment (runk) [inline test/data/migrations/20141028-aaa.js]: do it in new test migration 20150601-ddd.js please

File: test/data/migrations/20150601-ddd.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+async = require('async')
+
+
+exports.up = function(next) {
+  async.each([1,2,3,4], function(colName, cb) {
```
Comment (runk) [inline test/data/migrations/20150601-ddd.js]: `async.each([1, 2, 3, 4], function(n, cb) {`

File: test/migrant_test.js [hunk 5]
```diff
@@ -154,6 +158,13 @@ describe('index', function() {
         })
       })
     });
+
+    it('should recognise double callbacks', function(done){
+      m.migrate('up', 10, function(err, res) {
+        assert.equal(err, 'Error: The callback called more than once in 20150601-ddd.js')
+        done()
+      });
+    });
```
Comment (runk) [inline test/migrant_test.js]: :100:

**Comment by pphminions** [comment]
@runk Incorporated the code review comments

**Comment by runk** [comment]
:rocket:

**Pull request merged**